src_cpp/union_find.hpp

```diff
@@ -241,14 +255,28 @@ struct Cluster{
             this->spanning_tree_bits.insert(bit_index);
         }
 
+        // add the virtual boundary check
+        if(this->contains_boundary_bits == true){
+            // std::cout<<"Boundary check added"<<std::endl;
+            // std::cout<<"Contains boundary bits: "<<this->contains_boundary_bits<<std::endl;
+            this->check_nodes.insert(-1);
+        }
+
         for(int check_index: this->check_nodes){
             this->spanning_tree_check_roots[check_index] = check_index;
         }
 
+
+
         int check_neighbours[2];
         for(int bit_index: this->bit_nodes){
             check_neighbours[0] = this->pcm.column_heads[bit_index]->up->row_index;
             check_neighbours[1] = this->pcm.column_heads[bit_index]->down->row_index;
+
+            if(check_neighbours[0] == check_neighbours[1]){
```

Comment:
This handles the case of column weight 1 bits, right?

Reply:
Yes precisely. And adds a virtual boundary node.